pkg/api/util/conditions.go

```diff
@@ -249,6 +264,24 @@ func CertificateRequestReadyReason(cr *cmapi.CertificateRequest) string {
 	return ""
 }
 
+// This returns with the message if the CertificateRequest contains an
+// Failed condition, and returns "" otherwise.
+func CertificateRequestFailedMessage(cr *cmapi.CertificateRequest) string {
```

Comment:
Hm, whilst we do have `CertificateRequestReadyReason`, this is because it is utilised at multiple places within the codebase and makes readability better. In this case, we're only really using it in one place (here: https://github.com/jetstack/cert-manager/pull/2782/commits/eb16d018aa434a3fd6ee8727442060f32ca006d4#diff-528fa7c9a9cfce42c4813bd3207bb714R199)

Can we instead change this to be a call to `GetCertificateRequestCondition`, and then reference the appropriate within that accordingly? We can then remove this altogether 😄